### ISSUE TITLE
CSP without default-src is incorrectly detected as secure

### ISSUE THREAD

**Issue by Rob--W** [mozilla/addons-linter#4578]
### Describe the problem and steps to reproduce it:

The CSP validator's primary purpose is to reject CSP policies that permit remote code.
`script-src` is the primary directive to control that, and if that is not specified, `default-src`.
The validator correctly handles these two cases. But if `default-src` is missing, the default fallback is to be very permissive. This should be flagged as insecure CSP.

Test case, create `manifest.json` and lint the directory containing the following:
```json
{
  "name": "x",
  "version": "1",
  "manifest_version": 2,
  "content_security_policy": "img-src 'none'"
}
```

### What happened?
MANIFEST_CSP warning should be emitted, because this policy allows remote code.

### What did you expect to happen?
No warnings.

Note: Firefox has a different CSP parser. Its current behavior given the above CSP is to ignore it and use the default CSP that rejects remote scripts. While that matches our intent, the rejection of the CSP means that the extension developer's intent to reject images (`img-src 'none'`) is not respected.

### Anything else we should know?

This issue happens because [the implementation](https://github.com/mozilla/addons-linter/blob/a2fbda70d90106900d05be18410285afe22d13b9/src/parsers/manifestjson.js#L1045
) assumes that the CSP is secure by default, and rejects when an insecure directive is encountered. An empty CSP is not secure by default, which results in this bug.

**Comment by AlexandraMoga** [comment]
> ### What happened?
> MANIFEST_CSP warning should be emitted, because this policy allows remote code.
> 
> ### What did you expect to happen?
> No warnings.

@Rob--W I think the expected vs actual results are inverted in your STR.

The old behavior, as seen on prod currently, is to emit no warnings for the mentioned CSP:
![image](https://user-images.githubusercontent.com/31961530/202141473-a112dc06-876b-4711-b22d-df45fcdbb68b.png)

After the fix (linter 5.22.0), we are raising a warning, which should be the new expected behavior:
![image](https://user-images.githubusercontent.com/31961530/202142374-28c5e4cb-252b-42b0-86e7-484c05990f8d.png)

**Comment by Rob--W** [comment]
> > ### What happened?
> > MANIFEST_CSP warning should be emitted, because this policy allows remote code.
> > 
> > ### What did you expect to happen?
> > No warnings.
> 
> @Rob--W I think the expected vs actual results are inverted in your STR.

Thanks, I have edited the post and swapped the two. The verified behavior that you have observed is as desired.
